src/group_theory/iwasawa.lean

```diff
@@ -0,0 +1,655 @@
+-- import group_theory.subgroup.basic
+import group_theory.subgroup.pointwise
+import group_theory.coset
+import group_theory.quotient_group
+import group_theory.abelianization
+import group_theory.group_action.defs
+import group_theory.group_action.basic
+import group_theory.group_action.group
+import group_theory.group_action.conj_act
+
+
+/- 
+# Iwasawa criterion and simplicity of some groups
+
+Reference : https://www.math.ens.fr/~vanhaecke/tdalg2019/Correction_Iwasawa.pdf
+
+## Iwasawa criterion
+
+Let G be a group acting on a set X.
+
+**Definition**. The action is *transitive* if there is x:X whose orbit is X
+
+**Definition.** The action is *primitive* if it is transitive,
+and if all stabilizers G_x are maximal subgroups.
+
+**Lemma.** If the action is 2-transitive, then it is primitive.
+
+**Theorem (Iwasawa criterion).** Let G be a group with a primitive action on a set X.
+Assume given an Iwasawa structure, namely, 
+for all x:X, an abelian subgroup T x of G such that 
+1. ∀ g:G x:X, T (g x) = g (T x) g⁻¹ 
+2. The union of all T x generates G
+Then every normal subgroup of G that acts nontrivially on X contains
+the derived subgroup D(G) of G.
+
+**Corollary.** Let G be a perfect group (D(G)=G) 
+with a faithful and primitive action on a set X.
+Assume given an Iwasawa structure.
+Then G is simple.
+
+## Simplicity of special linear groups
+
+**Theorem.** Let K be a field and let n be an integer such that n ≥ 2.
+Unless n = 2 and K has ≤ 3 elements, the group PSL(n,K) is simple.
+
+
+## Simplicity of alternate groups
+
+**Theorem.** Let n be an integer such that n ≥ 5. 
+The alternate group A(n) is simple.
+
+-/
+
+
+section Maximal_Subgroups
+variables {G : Type*} [group G]
+
+/-- An subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+class subgroup.is_maximal (K : subgroup G) : Prop := 
+  (out : is_coatom K)  
+
+theorem is_maximal_def {K : subgroup G} : K.is_maximal ↔ is_coatom K := ⟨λ h, h.1, λ h, ⟨h⟩⟩
+
+theorem is_maximal.ne_top {K : subgroup G} (h : K.is_maximal) : K ≠ ⊤ := (is_maximal_def.1 h).1
+
+theorem is_maximal_iff {K: subgroup G} : K.is_maximal ↔
+  K ≠ ⊤ ∧ ∀ (H: subgroup G) g, K ≤ H → g ∉ K → g ∈ H → H = ⊤ :=
+begin
+  split, 
```

Comment:
Please put `{ }` around the subproofs that work on the two goals.
I see that you are structuring your proof with newlines, which also indicates the subproofs, but the mathlib style is to use `{ }`-blocks around subproofs.